### PR TITLE
[CALCITE-3591] Add bit_xor aggregate function

### PR DESCRIPTION
**Pull request by wangxlong**
Add BIT_XOR BIT aggregate function.
The discussion see [CALCITE-3591](https://issues.apache.org/jira/browse/CALCITE-3591)

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java [hunk 1]
```diff
@@ -1554,7 +1556,9 @@ public Expression implementResult(AggContext info,
       SqlAggFunction aggregation = info.aggregation();
       final Method method = (aggregation == BIT_AND
           ? BuiltInMethod.BIT_AND
-          : BuiltInMethod.BIT_OR).method;
+          : (aggregation == BIT_OR
```
Comment (ritesh-kapoor) [inline core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java]: It would be nice if this is simplified :)

Comment (amaliujia) [reply core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java]: How about just use if else or switch.

The current nested structure becomes less readable

Comment (hsyuan) [reply core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java]: Agree. Simple if else or switch should be fine.

Comment (wangxlong) [reply core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java]: Good idea. Thank you, I have updated.

File: core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java [hunk 2]
```diff
@@ -1080,6 +1080,13 @@ public static long bitOr(long b0, long b1) {
     return b0 | b1;
   }
 
+  // ^
+
```
Comment (jinxing64) [inline core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java]: We only have one implementation method for `^`, do we need this empty line ?

File: core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java [hunk 3]
```diff
@@ -1552,9 +1554,15 @@ public Expression implementResult(AggContext info,
...
-      final Method method = (aggregation == BIT_AND
-          ? BuiltInMethod.BIT_AND
-          : BuiltInMethod.BIT_OR).method;
+
+      BuiltInMethod builtInMethod = BuiltInMethod.BIT_AND;
+      if (aggregation == BIT_OR) {
+        builtInMethod = BuiltInMethod.BIT_OR;
+      } else if (aggregation == BIT_XOR) {
+        builtInMethod = BuiltInMethod.BIT_XOR;
+      }
```
Comment (jinxing64) [inline core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java]: Shall we do it like be below ?
```
if (aggregation  == BIT_AND) {
  ...
} else if (BIT_OR) {
  ...
} else if (BIT_XOR) {
  ...
} else {
  throw IllegalArgumentException
}
```
Or `Switch` is also cool for me.

File: core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java (hunk 2 above)
Comment (wangxlong) [reply core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java]: You are right, Thank you @jinxing64, I have updated

File: core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java (hunk 3 above)
Comment (wangxlong) [reply core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java]: Thanks for your advise @jinxing64, I have updated.

Comment (danny0405) [reply core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java]: Use switch case seems more clear.

Comment (wangxlong) [reply core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java]: @danny0405 Thank you, I have updated.

Comment (wangxlong) [reply core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java]: @danny0405 Thank you, I have updated.

**Pull request closed without merging**